### DISCUSSION TITLE
Creating authorized applications using discord.py

### DISCUSSION THREAD

**Discussion by BaiqingL** [Rapptz/discord.py#6898] [Q&A]
I am trying to write a terminal application interface for discord, and with my skillset discord.py is the most straightforward way. 

Since I am not creating a bot, and the fact that the discord TOS seems to disallow bot logging in as accounts, I want to ask what is the proper way to create an authorized application to log in as the user and read messages that way. 

For example, I have a logitech app in my "Authorized Apps" section that can have certain permissions. How would I do that in discord.py?

**Comment by chrisdewa** [comment]
There is none as far as i know. 

The posture of discord regarding this is very clear.

https://discord.com/developers/docs/topics/oauth2#bots

**Comment by Rapptz** [comment]
discord.py currently has no support for the OAuth2 API. Note however that even with OAuth2 you cannot read other user's messages. The closest thing you can get is the deprecated RPC message.read whitelisted scope but Discord no longer gives access to this.

**Reply by BaiqingL** [reply]
Hello, this might be a wording issue but essentially even if you made an OAuth2 application, had users clearly give consent and authenticate reading of messages, there still doesn't exist a current solution? Thanks.

**Reply by chrisdewa** [reply]
The user is not the issue, it's the use of the API
